packages/common/container-definitions/src/audience.ts

```diff
@@ -21,6 +20,56 @@ export interface IAudienceOwner extends IAudience {
 	 * @returns if a client was removed from the audience
 	 */
 	removeMember(clientId: string): boolean;
+
+	/**
+	 * Notifies Audience that current clientId has changed.
+	 * See {@link IAudience.getSelf} and {@link IAudienceEvents}'s "selfChanged" event for more details.
+	 */
+	setCurrentClientId(clientId: string): void;
+}
+
+/**
+ * @public
+ */
+export interface IAudienceEvents extends IEvent {
+	(
+		event: "addMember" | "removeMember",
+		listener: (clientId: string, client: IClient) => void,
+	): void;
+	/**
+	 * Notifies that client established new connection and caught-up on ops.
+	 * @param oldValue - represents old connection. Please note that oldValue.client in almost all cases will be undefined,
+	 * due to specifics how Audience refreshes on reconnect. In the future we could improve it and always provide client information.
+	 * @param newValue - represents newly established connection. While {@link IAudience.getSelf} is experimental, it's note guaranteed that
+	 * newValue.client is present. Same is true if you are consuming audience from container runtime layer and running against old version of loader.
+	 */
+	(event: "selfChanged", listener: (oldValue: ISelf | undefined, newValue: ISelf) => void): void;
+}
+
+/**
+ * Return type of {@link IAudience.getSelf}. Please see remarks for {@link IAudience.getSelf} to learn more details on promises.
+ * @public
+ */
+export interface ISelf {
+	/**
+	 * clientId of current or previous connection (if client is in disconnected or reconnecting / catching up state)
+	 * If client never connected to ordering service, then {@link IAudience.getSelf} will return undefined.
+	 * undefined if this client has never connected to the ordering service.
+	 * It changes only when client has reconnected, caught up with latest ops and certain other criteria are met.
+	 */
+	clientId: string;
+
+	/**
+	 * Information about current user, supplied by ordering service when client connected to it
+	 * and received clientId above.
```

Comment:
"Above" will lose its meaning when someone is reading this as docs in an IDE or the site.
```suggestion
	 * and received {@link ISelf.clientId}.
```